docs/concepts/workspaces.md

```diff
@@ -1,75 +1,170 @@
 # Workspaces
 
-Workspaces help organize large codebases by splitting them into multiple packages with independent
-dependencies. Each package in a workspace has its own `pyproject.toml`, but they are all locked
-together in a shared lockfile and installed to shared virtual environment.
+Inspired by the [Cargo](https://doc.rust-lang.org/cargo/reference/workspaces.html) concept of the
+same name, a workspace is "a collection of one or more packages, called _workspace members_, that
+are managed together."
 
-Using the project interface, `uv run` and `uv sync` will install all packages of the workspace,
-unless you select a single workspace member with `--package`. When using the `uv pip` interface,
-workspace dependencies behave like editable path dependencies.
+Workspaces organize large codebases by splitting them into distinct packages with independent
+dependencies. Think: a FastAPI-based web application, alongside a series of libraries that are
+versioned and maintained as separate Python packages, all in the same Git repository.
 
-## When (not) to use workspaces
+In a workspace, each package defines its own `pyproject.toml`, but the workspace shares a single
+lockfile, ensuring that the workspace operates with a consistent set of dependencies.
 
-One common use case for a workspace is that the codebase grows large, and eventually you want some
-modules to become independent packages with their own dependency specification. Other use cases are
-separating parts of the codebase with different responsibilities, e.g. in a repository with a
-library package and CLI package, where the CLI package makes features of the library available but
-has additional dependencies, a webserver with a backend and an ingestion package, or a library that
-has a performance-critical subroutine implemented in a native language.
+As such, `uv lock` operates on the entire workspace at once, while `uv run` and `uv sync` operate on
+the workspace root by default, though both accept a `--package` argument, allowing you to run a
+command _in_ any workspace member _from_ any workspace directory.
 
-Workspaces are not suited when you don't want to install all members together, members have
-conflicting requirements, or you simply want individual virtual environments per project. In this
-case, use regular (editable) relative path dependencies.
+## Getting started
 
-Currently, workspace don't properly support different members having different `requires-python`
-values, we apply the highest of all `requires-python` lower bounds to the entire workspace. You need
-to use a `uv pip` to install individual member in an older virtual environment.
+To create a workspace, add a `tool.uv.workspace` table to a `pyproject.toml`, which will implicitly
```

Comment:
We could also suggest in a !!! tip that `uv init child` in a project will do this for you.